src/common/connection.ts

```diff
@@ -524,6 +512,17 @@ export class Connection extends EventEmitter {
           }, retryTimeout)
         }
       })
+      // Finalize the connection and resolve all awaiting connect() requests
+      try {
+        this._retryConnectionBackoff.reset()
+        await this._subscribeToLedger()
+        this._startHeartbeatInterval()
+        this._connectionManager.resolveAllAwaiting()
+        this.emit('connected')
+      } catch (error) {
+        this._connectionManager.rejectAllAwaiting(error)
+        await this.disconnect().catch(() => {}) // Ignore this error
```

Comment:
Why do we ignore this error?

Reply:
I was mimicking code around ~407, but I think I was wrong. What should be done with the error? Can we emit the error, similar to an error on reconnect?

```await this.disconnect().catch((error) => {
      this.emit('error', 'disconnect', error.message, error)
})```

We should probably await and catch here as it's inside of the listener.

Reply:
And yes, what you summarized above is the problem.

Reply:
So the reason that it would be okay to ignore the error here is that we are already rejecting the `connect()` promise with the error on the line above: `this._connectionManager.rejectAllAwaiting(error)`

Ultimately, the problem here is: in a world where we have two errors, which do we return to the user when we can only reject one error? We could look into creating a new `MultipleError` type, but for now it feels like the failure to subscribe is more important than a failure to disconnect, which would make no sense to the user in the context of a failed `connect()`.

Reply:
Ok, it kind of seemed that way. I'll remove the emit of disconnect error then, as before.